src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/LanguageServices/CSharpLanguageServiceHost.cs

```diff
@@ -25,14 +26,11 @@ protected override Guid IntelliSenseProviderGuid
             get { return CSharpIntellisenseProvider; }
         }
 
-        /// <summary>
-        /// Invoked when the UnconfiguredProject is first loaded to initialize language services.
-        /// </summary>
-        [ProjectAutoLoad(ProjectLoadCheckpoint.InitialActiveConfigurationKnown)]
+        [ProjectAutoLoad(ProjectLoadCheckpoint.ProjectFactoryCompleted)]
         [AppliesTo(ProjectCapability.CSharpLanguageService)]
-        private void Initialize()
+        private Task OnProjectFactoryCompleted()
         {
-            var nowait = InitializeAsync();
+            return InitializeAsync();
```

Comment:
Not sure I understand - what would await do here - other than create an indirection?

How is:
``` C#
private Task OnProjectFactoryCompleted()
{
    return InitializeAsync();
}

private async Task InitializeAsync()
{
[...]
}
```

Any different from:

``` C#
private async Task OnProjectFactoryCompleted()
{
    return await InitializeAsync();
}

private async Task InitializeAsync()
{
[...]
}
```